core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -48,11 +48,13 @@ object TransactionCoordinator {
       config.transactionTransactionsExpiredTransactionCleanupIntervalMs)
 
     val producerIdManager = new ProducerIdManager(config.brokerId, zkUtils)
+    // we need to enable reaper thread still even with infinite delays for txn marker purgatory, since we
+    // are still relying on the thread to remove already completed tasks as each task is watched on two lists: txn id and partition id
```

Comment:
Updated the comments.

The reason we did not move it inside `TransactionMarkerChannelManager` is for convenient unit tests; otherwise we need to expose the purgatory in order to check its status.